src/VisualStudio/Core/Def/Implementation/ProjectSystem/DocumentProvider.cs

```diff
@@ -378,10 +380,25 @@ private void TryProcessOpenForDocCookie_NoLock(uint docCookie)
                         }
                         else
                         {
-                            TextBufferDataEventsSink.HookupHandler(this, shimTextBuffer, documentKey);
+                            TextBufferDataEventsSink.HookupHandler(shimTextBuffer, onDocumentLoadCompleted: () => OnDocumentLoadCompleted(shimTextBuffer, documentKey, moniker));
                         }
                     }
                 }
+
+                if (!hasAssociatedRoslynDocument && this._documentTrackingServiceOpt != null && !_docCookiesToNonRoslynDocumentBuffers.ContainsKey(docCookie))
+                {
+                    // Non-Roslyn document opened.
+                    var textBuffer = _editorAdaptersFactoryService.GetDocumentBuffer(shimTextBuffer);
+                    if (textBuffer != null)
+                    {
+                        this._documentTrackingServiceOpt.OnNonRoslynBufferOpened(textBuffer);
```

Comment:
I'd like this to be moved to it's own method and then called at the other location as well. Otherwise we're still risking a bug where we update one place but not the other.